### PR TITLE
Colony Guns Part 3: Longarms Rework

### PR DESCRIPTION
**Pull request by carlarctg**
# About the pull request

Updates and buffs the following weapons:

- Basira-Armstrong Rifle
- Model 12 Bolt-Action Rifle
- M37-17 pump shotgun
- Dragunov sniper rifle

### Basira-Armstrong Rifle

- The BA rifle has been reflavored as the removed 'ceremonial' ABR-40, now a hunting-civilian version of the L42. It effectively has the same stats as the L42, just don't take the stock off!
- Its magazines can only fit 12 bullets, but they still have the classic L42 kick to them. The magazines are completely compatible between both military and civilian gun types.
- Additionally, there are now holo-targeting magazines available for the ABR-40, for hunting target capture purposes.
- - The Contractor ERT has a chance to spawn with a tacticool ABR-40 loadout, with three spare holotarget magazines on their webbing.

### Model 12 Bolt-Action Rifle

- Like its sprite implies, this is now the new Basira-Armstrong rifle.
- Its stats have recieved an overhaul and it can now hold its own against a Xenomorph (or marine if you're CLF)
- Additionally, its bullets will push back (not stun) targets within three meters, to increase viability for colony usage.

### M37-17 Pump Shotgun

- Did you know that the 10% damage mod this gun was supposed to have didn't work? Now it does! And it deals 15% more damage to make up for it.
- However, it can now be melted down.

### Dragunov Sniper Rifle

- The dragunov has been revamped into a DMR, needing no skill to fire, dealing good damage, and having the same push-back feature the bolt-action now has.

# Explain why it's good for the game

All of these weapons are seen as, and are, a complete joke and ignored in favor of chungillionaire among us sus impostor running around PBing xenos with buckshot. Buffing them to be useful paves the way for civilians to use civilian weaponry instead of chungillionarius.

### ABR-40 Hunting Rifle

This is intended as asset recycling. IIRC, Triiodine disliked the existence of a ceremonial rifle and thought that was goofy, which is an understandable opinion, but it pains me to see the gun's cool assets go unused. I took the opportunity and made the lame and mediocre Basira into this cool gun which marines and survivors will take a lot more interest in than a Basira plinker.

While it does have L42 stats, colonists won't be able to locate any L42A ammo on the colony, meaning they are limited to 12-round magazines. Still, giving them a weapon like this is a great way to incentivize them to step out of their chungus zone. 

As for the contractor addition, these are supposed to be ex-USCM mercenaries, who are trained in the usage of marine equipment. It makes sense that they'd grab the civilian version of a primary marine gun and tune it to their liking.

### Basira-Armstrong Bolt-Action Hunting Rifle

Chomp made this really cool backend for a bolt-action that tragically ended up never being used, not really, aside from a half-hearted few rifles being thrown in at Shiva's Snowball. Since these guns are so weak, it's plain to see why nobody even looks at them twice. So I changed that.

### M37-17 Pump Shotgun

Bugfix and a small damage buff to make up for it. It being unacidable was lame anyways. Spread projectiles are still bugged and don't inherit the base gun's damage mod, but that's out of scope.

### Dragunov Designated Marksman Rifle

This gun has been a joke since 2017, it's time to give it some love. Still needs one-hand sprites but not my problem.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
balance: Updates and buffs the following weapons: Basira-Armstrong Rifle, Model 12 Bolt-Action Rifle, M37-17 pump shotgun, Dragunov sniper rifle
imageadd: The Basira rifle has been reflavored as the removed 'ceremonial' ABR-40, now a hunting-civilian version of the L42. It effectively has the same stats as the L42, just don't take the stock off!
add: Its magazines can only fit 12 bullets, but they still have the classic L42 kick to them. The magazines are completely compatible between both military and civilian gun types.
add:: Additionally, there are now holo-targeting magazines available for the ABR-40, for hunting target capture purposes.
add: The Contractor ERT has a chance to spawn with a tacticool ABR-40 loadout, with three spare holotarget magazines on their webbing.
balance: Like the bolt-action rifle's sprite implies, this is now the new Basira-Armstrong rifle.
add: Its stats have recieved an overhaul and it can now hold its own against a Xenomorph (or marine if you're CLF)
add: Additionally, its bullets will push back (not stun) targets within three meters, to increase viability for colony usage.
balance: Did you know that the 10% damage mod the M37-17 pump shotgun was supposed to have didn't work? Now it does! And it deals 15% more damage to make up for it.
del: However, it can now be melted down.
balance: The dragunov has been revamped into a DMR, needing no skill to fire, dealing good damage, and having the same push-back feature the bolt-action now has.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/rifles.dm [hunk 1]
```diff
@@ -1330,54 +1330,84 @@
...
-	item_state = "hunting"
-	reload_sound = 'sound/weapons/handling/l42_reload.ogg'
-	unload_sound = 'sound/weapons/handling/l42_unload.ogg'
-	fire_sound = 'sound/weapons/gun_carbine.ogg'
-	current_mag = /obj/item/ammo_magazine/rifle/hunting
+// Civilian version of the L42A, used for hunting, and also by undersupplied paramilitary groups.
+
+/obj/item/weapon/gun/rifle/l42a/abr40
+	name = "\improper ABR-40 hunting rifle"
+	desc = "The ABR-40 was created after the striking popularity of the L42 battle rifle as a hunting rifle for civilians. Sporting faux wooden furniture and a legally-mandated 12 round magazine, it's still highly accurate and deadly, a favored pick of experienced hunters and retired Marines. However, it's very limited in attachment selection, only being able to fit rail attachments, and the differences in design from the L42 force an awkward pose when attempting to hold it one-handed. Removing the stock is not recommended."
```
Comment (stanalbatross) [inline code/modules/projectiles/guns/rifles.dm:1344]: too long, put most of this into a lore desc

File: code/modules/projectiles/guns/rifles.dm [hunk 2]
```diff
@@ -1330,54 +1330,84 @@
...
-	damage_mult = BASE_BULLET_DAMAGE_MULT + BULLET_DAMAGE_MULT_TIER_6
-	recoil_unwielded = RECOIL_AMOUNT_TIER_4
-	damage_falloff_mult = 0
+	accuracy_mult = (BASE_ACCURACY_MULT + HIT_ACCURACY_MULT_TIER_5) - HIT_ACCURACY_MULT_TIER_10
+	recoil = RECOIL_AMOUNT_TIER_4
+	scatter = (SCATTER_AMOUNT_TIER_8) + SCATTER_AMOUNT_TIER_5
+
+
+/obj/item/weapon/gun/rifle/l42a/abr40/tactical
+	desc = "The ABR-40 was created after the striking popularity of the L42 battle rifle as a hunting rifle for civilians, and naturally fell into the hands of many underfunded paramilitary groups and insurrections in turn, especially due to its cross-compatibility with L42A magazines. This specific rifle seems to have unique tacticool blue-black furniture alongside some miscellaneous aftermarket modding."
```
Comment (stanalbatross) [inline code/modules/projectiles/guns/rifles.dm:1380]: same

File: code/modules/projectiles/guns/shotguns.dm [hunk 3]
```diff
@@ -1115,12 +1111,15 @@ can cause issues with ammo types getting mixed up during the burst.
...
 	name = "\improper M37-17 pump shotgun"
-	desc = "A ten-round pump action shotgun with five-round capacity dual internal tube magazines allowing for quick reloading and highly accurate fire. Issued to select USCM vessels out on the rim. You can switch the active internal magazine by toggling burst fire mode."
+	desc = "A military version of the iconic HG 37-12, this design can fit one extra shell in each of its dual-tube internal magazines, and fires shells with increased velocity, resulting in more damage. Issued to select USCM vessels out on the rim. You can switch the active internal magazine by toggling burst fire mode."
 	icon_state = "m3717"
 	item_state = "m3717"
 	current_mag = /obj/item/ammo_magazine/internal/shotgun/cmb/m3717
 	starting_attachment_types = list(/obj/item/attachable/stock/hg3712/m3717)
-	unacidable = TRUE
-	damage_mult = 1.1
+	unacidable = FALSE
```
Comment (stanalbatross) [inline code/modules/projectiles/guns/shotguns.dm:1119]: shouldn't this be inherited

**Review by stanalbatross — CHANGES_REQUESTED** [review]
and drop all changes to the type 23

File: code/modules/projectiles/guns/rifles.dm (hunk 1 above)
Comment (carlarctg) [reply code/modules/projectiles/guns/rifles.dm]: yea

File: code/modules/projectiles/guns/rifles.dm (hunk 2 above)
Comment (carlarctg) [reply code/modules/projectiles/guns/rifles.dm]: yea

File: code/modules/projectiles/guns/shotguns.dm (hunk 3 above)
Comment (carlarctg) [reply code/modules/projectiles/guns/shotguns.dm]: yea

File: code/modules/projectiles/guns/rifles.dm [hunk 4]
```diff
@@ -1330,54 +1330,86 @@
...
-	damage_mult = BASE_BULLET_DAMAGE_MULT + BULLET_DAMAGE_MULT_TIER_6
-	recoil_unwielded = RECOIL_AMOUNT_TIER_4
-	damage_falloff_mult = 0
+	accuracy_mult = (BASE_ACCURACY_MULT + HIT_ACCURACY_MULT_TIER_5) - HIT_ACCURACY_MULT_TIER_10
+	recoil = RECOIL_AMOUNT_TIER_4
+	scatter = (SCATTER_AMOUNT_TIER_8) + SCATTER_AMOUNT_TIER_5
+
+
+/obj/item/weapon/gun/rifle/l42a/abr40/tactical
+	desc = "The civilian version of the L42A battle rifle that is often wuielded by Marines. Almost identical and even cross-compatible with L42 magazines, just don't take the stock off. This rifle seems to have unique tacticool blue-black furniture alongside some miscellaneous aftermarket modding."
```
Comment (carlarctg) [inline code/modules/projectiles/guns/rifles.dm:1381]:
```suggestion
	desc = "The civilian version of the L42A battle rifle that is often wielded by Marines. Almost identical and even cross-compatible with L42 magazines, just don't take the stock off. This rifle seems to have unique tacticool blue-black furniture alongside some miscellaneous aftermarket modding."
```

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Comment by stanalbatross** [comment]
approval is as always consistent on them not being cracked

**Review by stanalbatross — APPROVED** [review]
needs sprite and map approval :3

**Review by kugamo — APPROVED** [review]
Sprites

**Pull request merged**